### PR TITLE
S/green build comment

### PR DESCRIPTION
**Pull request by stevenweber**
https://trello.com/c/yKFeSlyp/70-add-ci-review-to-emergency-ci-prs

On a daily cron, this will

* find any prs that have bypassed ci checks and have not been verified
* find a green build for master
* tag those pr as verified and add a comment with the green build

**Goodies**

This adds a `run-development` rule that allows you to actually run the action locally!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -99,5 +99,10 @@ __tests__/runner/*
 lib/**/*
 
 test.js
+tmp/
```
Comment (sbowenwilliams) [inline .gitignore:102]: <3

File: __tests__/circle_ci.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      });
+
+      const pipelineNumber = 1234;
+      const workflowPath = `/api/v2/workflow/${id}`;
+      nock(API).get(workflowPath).reply(200, {
+        pipeline_number: pipelineNumber,
+      });
+
+      const workflow = await fetchMostRecentGreenWorkflow();
+      expect(workflow).toEqual(expect.objectContaining({
```
Comment (sbowenwilliams) [inline __tests__/circle_ci.test.ts:55]: TIL about `expect.objectContaining`

**Comment by ebenoist** [comment]
> This adds a run-development rule that allows you to actually run the action locally!

Will this still be needed if adopt these new features?

This adds a run-development rule that allows you to actually run the action locally!

File: .github/workflows/sample.yml [hunk 3]
```diff
@@ -28,6 +31,8 @@ jobs:
         github_token: ${{ secrets.GITHUB_TOKEN }}
         relevant_labels: "console-access"
         required_checks: "ci/circleci: test,ci/circleci: lint"
+        circle_ci_token: ${{ secrets.CIRCLE_CI_TOKEN }}
+        circle_ci_workflow_name: 'build_and_test'
```
Comment (ebenoist) [inline .github/workflows/sample.yml:35]: Does this need to be different from the required checks above?

File: Makefile [hunk 4]
```diff
@@ -25,5 +32,22 @@ test: ## verify functionality
 test-watch: ## start test watcher
 	jest --watch --watchPathIgnorePatterns tmp
 
+${ENV_FILE}:
```
Comment (ebenoist) [inline Makefile:35]: I've never seen targets use curlies before. Do these have a different semantic?

File: src/circle_ci.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  id: string;
+  name: string;
+  pipeline_id: UUID;
+  pipeline_number: number;
+  project_slug: string;
+  status: Status;
+  stopped_at: string;
+}
+
+async function request<R>(
```
Comment (ebenoist) [inline src/circle_ci.ts:60]: Why can't we just look at the GitHub repository status checks like the other breakglass workflows?

**Comment by ebenoist** [comment]
I'd probably prefer not pulling this status from circle-ci if we can avoid it? It creates yet another system dependency that I think can be replaced with a GitHub status check on master? This would also keep the "required" checks inline with the posthoc review.

File: run-development.example.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// this is a stub for running one-off commands to try things out in development.
+// use the .env file to customize action inputs, see readme for more details.
+// usage: make run-test
+
+import {
```
Comment (sbowenwilliams) [inline run-development.example.ts:5]: I love this

File: Makefile (hunk 4 above)
Comment (stevenweber) [reply Makefile:35]: In this context I believe there's any difference.

File: .github/workflows/sample.yml (hunk 3 above)
Comment (stevenweber) [reply .github/workflows/sample.yml:35]: Yeah I can check the status based on required checks in github.

**Comment by stevenweber** [comment]
> > This adds a run-development rule that allows you to actually run the action locally!
> 
> Will this still be needed if adopt these new features?
> 
> https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

The `run-development` triggers code execution in an environment that matches how actions are run whether github-hosted or self-hosted. Not sure I fully follow your thought, but if we built-out self-hosted runners to be run locally for development, you would still need some trigger in-lieu of a rule like this.

**Pull request merged**